include/zephyr/toolchain/gcc.h

```diff
@@ -489,85 +489,52 @@ do {                                                                    \
  * to generate named symbol/value pairs for kconfigs.
  */
 
-#if defined(CONFIG_ARM)
-
-/*
- * GNU/ARM backend does not have a proper operand modifier which does not
- * produces prefix # followed by value, such as %0 for PowerPC, Intel, and
- * MIPS. The workaround performed here is using %B0 which converts
- * the value to ~(value). Thus "n"(~(value)) is set in operand constraint
- * to output (value) in the ARM specific GEN_OFFSET macro.
+#if defined(CONFIG_ARM) || defined(CONFIG_X86) || defined(CONFIG_ARC) || defined(CONFIG_ARM64) ||  \
+	defined(CONFIG_NIOS2) || defined(CONFIG_XTENSA) || defined(CONFIG_MIPS) ||                 \
+	defined(CONFIG_ARCH_POSIX) || defined(CONFIG_SPARC)
+
+#define GEN_ABSOLUTE_SYM(name, value)                                                              \
+	do {                                                                                       \
+		__asm__(".global " #name);                                                         \
+		__asm__(".set " #name ", %c0" ::"n"(value));                                       \
+		__asm__(".type " #name ", STT_OBJECT");                                            \
+	} while (false)
+
+#define GEN_ABSOLUTE_SYM_KCONFIG(name, value)                                                      \
+	do {                                                                                       \
+		__asm__(".global " #name);                                                         \
+		__asm__(".set " #name ", " #value);                                                \
+		__asm__(".type " #name ", STT_OBJECT");                                            \
+	} while (false)
+
+/* The following is a workaround for the RISC-V target, which
+ * has a bug so it errors out on the target-agnostic '%c'.
+ *
+ * https://gcc.gnu.org/bugzilla/show_bug.cgi?id=112434
+ *
+ * > error: invalid 'asm': invalid use of '%c'
+ *
+ * According to commit cd83e85edc5d741f6b52c6b5995303c30bda443a,
```

Comment:
so it works now for NIOS II? It was under the same umbrella [in that commit](https://github.com/zephyrproject-rtos/zephyr/commit/cd83e85edc5d741f6b52c6b5995303c30bda443a#diff-186f761dc94b00ce3a8e23d41d4f87187cc4f0bbaa83a1bf8402401755c60dc2R136)

Reply:
Yes, I ran twister on the hello_world sample with `--build-only --all`. There is no difference in the output before and after applying this patch. All nios testsuites pass.

Reply:
Finally noticed this same workaround for a pretty different error message...

I searched the GAS manual but I couldn't find where these prefixes are documented. What difference does this 'c' make? It makes none in all the Xtensa toolchains I tried (when it does not crash).